base_external_api_ftp/models/apicli_connection_ftp.py

```diff
@@ -18,27 +18,36 @@ class ApicliConnection(models.Model):
         selection_add=[("ftp", "FTP")], ondelete={"ftp": "set default"}
     )
 
-    def delete_ftp_file(self, ftp_session, subdirectory):
+    def create_delete_file_message(self, ftp_session, subdirectory):
         from_path, subdir_list, file_list = next(os.walk(subdirectory or "/tmp"))
         for file_name in file_list:
             file_path = os.path.join(from_path, file_name)
-            state = "draft"
+            message_id = self.env["apicli.message"].create(
+                {
+                    "connection_id": self.id,
+                    "endpoint": file_name,
+                    "content": "",
+                    "state": "draft",
+                }
+            )
             try:
                 delete_msg = ftp_session.delete(file_name)
-                state = "done"
+                message_id.write(
+                    {
+                        "content": delete_msg,
```

Comment:
delete_msg is not the file content, it is the delete action response. So we're missing the file content.